### PR TITLE
add a deprecations and removals guide

### PR DESCRIPTION
**Pull request by timmysilv**
Link to the doc: https://xanaduai-pennylane--5083.com.readthedocs.build/en/5083/development/guide/deprecations_removals.html

**Context:**
There are a bunch of things to consider when deprecating code, so I thought I'd consolidate my process in a document for posterity.

**Description of the Change:**
Adding a step-by-step guide to deprecations and removals to the public developer docs.

**Benefits:**
PennyLane maintainers will know how we deprecate code forevermore

**Possible Drawbacks:**
what if this document itself gets out-of-date? 😨 

[sc-54330]

### CODE REVIEW

File: doc/development/guide/deprecations_removals.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+   opposed to deprecation.
+
+3. If the feature has public-facing docs, include a similar warning message in a visible part of
+   its docstring using a ``.. warning::`` Sphinx directive.
+
+4. Replace all uses of the deprecated code within PennyLane's own source code. This is to ensure
+   that the warning isn't unexpectedly raised for users that did not personally call the deprecated
+   piece of code.
+
+5. Remove any tests that specifically test the deprecated code. For tests that depend it but are
```
Comment (trbromley) [inline doc/development/guide/deprecations_removals.rst:42]:
```suggestion
5. Remove any tests that specifically test the deprecated code. For tests that depend on it but are
```

File: doc/development/guide/deprecations_removals.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+   not written to test it specifically, update them to use the new/preferred code.
```
Comment (trbromley) [inline doc/development/guide/deprecations_removals.rst:43]: Should we be removing the tests at the deprecation stage, or the removal stage?

**Review by trbromley — COMMENTED** [review]
Thanks @timmysilv, this is a great guide!

File: doc/development/guide/deprecations_removals.rst (hunk 2 above)
Comment (timmysilv) [reply doc/development/guide/deprecations_removals.rst]: it's probably wiser to wait until it's removed, will update

**Review by albi3ro — APPROVED** [review]
Thanks for putting this together 🎉

**Comment by trbromley** [comment]
Thanks @timmysilv! Might also be nice to include this in the release notes (the docs section).

**Comment by timmysilv** [comment]
👍 sneaking it in with #5094

**Pull request merged**